### PR TITLE
Fix build errors in MSVC (Windows)

### PR DESCRIPTION
**Pull request by Cidana-Developers**
1. Rename SvtAv1Enc output to "SvtAv1EncShared" to avoid
   name conflict with static library build.
2. Fix build break due to missing "EbFinishTime"
   and "EbComputeOverallElapsedTime".
3. Add -DBUILD_TESTING=ON in appveyor.yml to build unittest.

### CODE REVIEW

**Comment by hassount** [comment]
@tianjunwork @jsunintel 
any issues with this when linking to ffmpeg / gstreamer ?

**Comment by jsunintel** [comment]
> 
> 
> @tianjunwork @jsunintel
> any issues with this when linking to ffmpeg / gstreamer ?

We either build the SVT AV1 lib as static or shared, never both. Why changed the lib name?

And why are there two set of EbTime.h/EbTime.c under Lib/Common/Codec and App/EncApp seperately, with almost the same code/API in them?

**Comment by Cidana-Developers** [comment]
> > @tianjunwork @jsunintel
> > any issues with this when linking to ffmpeg / gstreamer ?
> 
> We either build the SVT AV1 lib as static or shared, never both. Why changed the lib name?
> 
static lib is always built, and shared libs is built if BUILD_SHARED_LIBS is enabled.  Following code is from Source/Lib/Encoder/Codec/CMakeLists.txt:
```
if(BUILD_SHARED_LIBS)
    # Encoder Lib Source Files
    add_library(SvtAv1EncShared SHARED
        ${all_files}
        $<TARGET_OBJECTS:COMMON_CODEC>
        $<TARGET_OBJECTS:COMMON_C_DEFAULT>
        $<TARGET_OBJECTS:COMMON_ASM_SSE2>
        $<TARGET_OBJECTS:COMMON_ASM_SSSE3>
        $<TARGET_OBJECTS:COMMON_ASM_SSE4_1>
        $<TARGET_OBJECTS:COMMON_ASM_AVX2>
    )
    set_target_properties(SvtAv1EncShared PROPERTIES OUTPUT_NAME SvtAv1EncShared)
    target_link_libraries(SvtAv1EncShared ${PLATFORM_LIBS})
    install(TARGETS SvtAv1EncShared DESTINATION "${CMAKE_INSTALL_LIBDIR}")
endif(BUILD_SHARED_LIBS)

add_library(SvtAv1EncStatic STATIC
    ${all_files}
    $<TARGET_OBJECTS:COMMON_CODEC>
    $<TARGET_OBJECTS:COMMON_C_DEFAULT>
    $<TARGET_OBJECTS:COMMON_ASM_SSE2>
    $<TARGET_OBJECTS:COMMON_ASM_SSSE3>
    $<TARGET_OBJECTS:COMMON_ASM_SSE4_1>
    $<TARGET_OBJECTS:COMMON_ASM_AVX2>
)
```
On windows platform, the lib names for static and shared are the same and the libs are in the same folder. In this case, the static lib will overwrite the shared lib, which will break the unittest build.

If static lib is not required by ffmpeg/gstreamer,  we can build shared lib only and do not change the lib name.

**Comment by tianjunwork** [comment]
Hi @Cidana-Developers, currently, gstreamer plugin links svt static lib. Can you make the lib type configurable?

**Comment by Cidana-Developers** [comment]
> Hi @Cidana-Developers, currently, gstreamer plugin links svt static lib. Can you make the lib type configurable?

Sure, will do it.

**Comment by Cidana-Developers** [comment]
@tianjunwork Done. Now either shared lib or static lib is built, which is controlled by BUILD_SHARED_LIBS. It will not break the ffmpeg/gstreamer build, since static lib is built by default.
@1480c1 Please review the CL too, thanks.

File: Source/Lib/Decoder/Codec/CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,17 @@ endif()
 
 if(BUILD_SHARED_LIBS)
     # Decoder Lib Source Files
-    add_library(SvtAv1DecShared SHARED
+    add_library(SvtAv1Dec SHARED
```
Comment (1480c1) [inline Source/Lib/Decoder/Codec/CMakeLists.txt]: This extra add_library is unecessary, just do add_library once without SHARED or STATIC and cmake will autochoose based on BUILD_SHARED_LIBS

File: Source/Lib/Encoder/Codec/CMakeLists.txt [hunk 2]
```diff
@@ -42,10 +42,20 @@ file(GLOB all_files
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     list(APPEND PLATFORM_LIBS m)
 endif()
-
+message(STATUS "BUILD_SHARED_LIBS : " ${BUILD_SHARED_LIBS})
+# Encoder Lib Source Files
 if(BUILD_SHARED_LIBS)
-    # Encoder Lib Source Files
-    add_library(SvtAv1EncShared SHARED
+    add_library(SvtAv1Enc SHARED
```
Comment (1480c1) [inline Source/Lib/Encoder/Codec/CMakeLists.txt]: Ditto here

File: Source/Lib/Decoder/Codec/CMakeLists.txt [hunk 3]
```diff
@@ -53,23 +63,10 @@ if(BUILD_SHARED_LIBS)
...
-    $<TARGET_OBJECTS:COMMON_C_DEFAULT>
-    $<TARGET_OBJECTS:COMMON_ASM_SSE2>
-    $<TARGET_OBJECTS:COMMON_ASM_SSSE3>
-    $<TARGET_OBJECTS:COMMON_ASM_SSE4_1>
-    $<TARGET_OBJECTS:COMMON_ASM_AVX2>
-)
-set_target_properties(SvtAv1DecStatic PROPERTIES OUTPUT_NAME SvtAv1Dec)
-target_link_libraries(SvtAv1DecStatic ${PLATFORM_LIBS})
-install(TARGETS SvtAv1DecStatic DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+set_target_properties(SvtAv1Dec PROPERTIES OUTPUT_NAME SvtAv1Dec)
```
Comment (1480c1) [inline Source/Lib/Decoder/Codec/CMakeLists.txt]: Also unecessary, output name is already set to it

File: Source/Lib/Encoder/Codec/CMakeLists.txt [hunk 4]
```diff
@@ -54,23 +64,10 @@ if(BUILD_SHARED_LIBS)
...
-    $<TARGET_OBJECTS:COMMON_C_DEFAULT>
-    $<TARGET_OBJECTS:COMMON_ASM_SSE2>
-    $<TARGET_OBJECTS:COMMON_ASM_SSSE3>
-    $<TARGET_OBJECTS:COMMON_ASM_SSE4_1>
-    $<TARGET_OBJECTS:COMMON_ASM_AVX2>
-)
-set_target_properties(SvtAv1EncStatic PROPERTIES OUTPUT_NAME SvtAv1Enc)
-target_link_libraries(SvtAv1EncStatic ${PLATFORM_LIBS})
-install(TARGETS SvtAv1EncStatic DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+set_target_properties(SvtAv1Enc PROPERTIES OUTPUT_NAME SvtAv1Enc)
```
Comment (1480c1) [inline Source/Lib/Encoder/Codec/CMakeLists.txt]: Ditto

Comment (Cidana-Developers) [reply Source/Lib/Encoder/Codec/CMakeLists.txt]: Done.

File: Source/Lib/Decoder/Codec/CMakeLists.txt (hunk 3 above)
Comment (Cidana-Developers) [reply Source/Lib/Decoder/Codec/CMakeLists.txt]: Done.

File: Source/Lib/Encoder/Codec/CMakeLists.txt (hunk 2 above)
Comment (Cidana-Developers) [reply Source/Lib/Encoder/Codec/CMakeLists.txt]: Done.

File: Source/Lib/Decoder/Codec/CMakeLists.txt (hunk 1 above)
Comment (Cidana-Developers) [reply Source/Lib/Decoder/Codec/CMakeLists.txt]: Done.

**Pull request merged**